Add Product Filter component to the new pricing page

vikneshwar opened the pull request:
#### Proposed Changes

* This PR adds a new `Products` and `ProductFilter` component inside `ProductStore`

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Boot up this PR 
    * Run `git fetch && git checkout add/product-filter-component`
    * Run `yarn start-jetpack-cloud`
    * Goto [http://jetpack.cloud.localhost:3001/pricing?flags=jetpack/pricing-page-rework-v1](http://jetpack.cloud.localhost:3001/pricing?flags=jetpack/pricing-page-rework-v1)
* Or click on Jetpack Cloud live link below
    * Goto `/pricing?flags=jetpack/pricing-page-rework-v1`
* Confirm that you see the Products and Bundles Filter in the top of the page 
* Confirm that the design matches the Figma design
* Switch between the filter and make sure the respective component loads 

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

#### Desktop Screenshot
![image](https://user-images.githubusercontent.com/2027003/186350856-f6bd8d30-60ec-4815-9d0f-b54e56ea3e14.png)


### Mobile Screenshot 
![image](https://user-images.githubusercontent.com/2027003/186350639-639d98a0-9512-48f4-bdd7-09c90b9a228e.png)



Completes: [1202796695664022-as-1202796695664061/f](1202796695664022-as-1202796695664055/f)
Figma: m0y1jHTuDpvMPkFsOJBoNP-fi-3715%3A132855

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202796695664061

vikneshwar committed:
Add Product and Product Filter component for the new pricing page

- Create a new product filter component
- Create a new product component
- Wire both the components to the main page

vikneshwar committed:
Add test cases for Products and ProductFilter component in the new pricing page

manzoorwanijk commented on client/my-sites/plans/jetpack-plans/product-store/product-filter.tsx:
```diff
@@ -0,0 +1,29 @@
+
+	return (
+		<div className="jetpack-product-store__product-filter">
+			<SegmentedControl className="jetpack-product-store__product-filter-toggle" compact primary>
```
`toggle` is a modifier here, so we can use this
```suggestion
			<SegmentedControl className="jetpack-product-store__product-filter--toggle" compact primary>
```

manzoorwanijk reviewed (commented):
Works as expected and looks good. [I yesterday learned](https://github.com/Automattic/wp-calypso/pull/66857#discussion_r952632521) from @jkguidaven about CSS standards recommending non-nested CSS. So I think let us avoid it.

Also, I would like @andrii-lysenko to have a look as well.

vikneshwar committed:
Product filter component fix styling

- remove nesting of css in product filter component
- adjust the top margin of the component

manzoorwanijk reviewed (approved):
This looks good now. Thank you for adding unit tests. I think let us wait for unit tests to be added for now, because the behavior and structure of the components may change over the course of the project. We can add unit tests when the components are in their final shape.
